README.md

```diff
@@ -1 +1,368 @@
+
+## 1.2 Inline Links
+
+IPLD Inline Links solve for the above by adding a standard way to inline data in an IPLD document without altering its hash. Link configuration (hashing algorithm, data encoding) is inherited from the surrounding context, unless the link is present. Explicitly including the CID to provide this configuration is OPTIONAL.
+
+Inline links do not modify the IPLD data model itself, but rather extend the format in a way that can be round-tripped to and from standard IPLD as desired.
+
+# 2 Format
+
+The basic structure of an inline link is similar to how reference links are represented in the most common IPLD codecs: signalling a link by wrapping the payload in a map with the single `"/"` key. By convention, this signals an IPLD-specific extension to a base codec.
+
+## 2.1 Embed Wrapper
+
+An inline link MUST be signalled by wrapping in a map with a `"/"` key.
+
+``` ipldsch
+type EmbedWrapper struct {
+  embed InlineLink (rename "/")
+}
+```
+
+## 2.2 Embedded IPLD Payload
+
+The inlined DAG payload MUST contain the inlined DAG. The CID MAY be present or set to `Null`.
+
+``` ipldsch
+type Embed struct {
+  link nullable Link
+  data          Any
+}
+```
+
+### 2.2.1 Explicit
+
+``` js
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "link": "bafyreif7dowvi5nuzzijawl22vpqsughufapj455diyflrk7htswzbjid4", // DAG-CBOR & SHA2-256
+      "data": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+```
+
+Explicit encoding and hash algorithm work as normal: the CID MAY be given in advance via the `"link"` field, and MUST validly describe the `"data"` field. 
+ 
+### 2.2.2 Inherited
+
+``` js
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "link": null, // Inherits encoding & hash algorithm from container
+      "data": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+```
+
+Inherited encoding is helpful in many situations. A core intention of IPLD is to abstract away from the exact data encoding. When including a link in IPLD, a hash algorithm and encoding MUST be hardcoded. Such choices are often premature, and lead to awkward design choices to avoid this encoding, or exceptions in prose.
+
+Inheriting from the surrounding context enables a level of flexibility for the nested data. Its hash and encoding MUST be the same as the parent, which is by far the most common case. In order to calculate the hash of the entire structure, the direct parent's encoding MUST be known, then passed to the encoder for the nested graph, the CID calculated, and placed into the parent. See more in [IPLD Canonicalization].
+
+The implicit strategy MAY lead to cases where the same DAG generates different CIDs in the same structure, based on different parent encoding contexts.
+
+## 2.3 Capsule
+
+To signal that a payload contains inlined links, the following capsule MAY be used:
+
+``` ipldsch
+type InlineLinkCapsule struct {
+  c Any (rename "ipld/inline/v0.1")
+}
```

Comment:
If we adopt `{ "/": { ".": ... } }` we probably won't need a separate capsule either, although `.` maybe not descriptive enough.

Reply:
Yeah agreed. Having an outermost capsule can be nice to signal that this is in our special format, but capsules are always optional.